### DISCUSSION TITLE
HID Recieve OUT Endpoint Confusion

### DISCUSSION THREAD

**Discussion by samsamm777** [hathach/tinyusb#1204] [Q&A]
I'm a little confused on how the `tud_hid_set_report_cb` should be working, when in relation to receiving reports on the OUT endpoint. The callback fires just fine when I send a report from the host. Good. The buffer length is also correct. However when it comes to reading the payload, ie any data after the first byte (reportID), its not there. I just get 0 on the 2nd byte.

The report buffer should simply be `0x2, 0x4` which is seems is correct in the buffer variable watcher. But not in the actual buffer array. Im just a bit confused. Anyone have any advice? Hopefully screenshot will help.

![hmm](https://user-images.githubusercontent.com/1075392/142766129-45f62b33-70a2-4138-83c7-3fef1565ceb8.PNG)

Thanks!

**Comment by hathach** [comment]
can you provide
- PC OS, any tools you use
- hardware: which mcu, board
- firmware: which example you are running. If it is not stock one, then make one based on them
- how to reproduce: which exactly did you do to send the report on host side (command/tools ?)

**Comment by chegewara** [comment]
Sorry for my 2 cents, but:
1. you have `report_id` in callback, and it is `0` on screenshot
2. `buffer[0]`, which you assign to `reportId` is in fact 0x20, not 0x02
3. report type is `HID_REPORT_TYPE_INVALID`
